data.py

```diff
@@ -0,0 +1,83 @@
+import csv
+import json
+import operator
+
+import nlu
+import questions
+
+
+class Similarity(object):
+    def __init__(self, intent: str, score: float):
+        self.intent = intent
+        self.score = score
+
+    def __repr__(self):
+        return '<Similarity {intent} {score}>'.format(
+            intent=self.intent, score=self.score)
+
+
+class SimilarityResult(object):
+
+    def __init__(self, query_text: str, intents: [Similarity]):
+        self.query = query_text
+        self.intents = intents
+
+    def __repr__(self):
+        return '<SimilarityResult query {query} topScoringIntent {topScoringIntent} intents {intents}>'.format(
+            query=self.query, topScoringIntent=self.topScoringIntent, intents=self.intents)
+
+    @property
+    def intents(self):
+        return self._intents
+
+    @intents.setter
+    def intents(self, intents: [Similarity]):
+        self._intents = intents
+        self.topScoringIntent = max(
+            self.intents, key=operator.attrgetter('score'))
+
+    def to_json(self):
+        return json.dumps(
+            self,
+            default=lambda o: o.__dict__,
```

Comment:
I would say remove the setter magic, what's the problem of `set_intent`?

Don't you think directly accessing something is start with `__` must be something wrong?

Reply:
I guess I know why you use the getter/setter, but I said, the cache won't perform as good as you thought! At the most time, we use it once, if you never use it, then caching can be even worse.